main.js

```diff
@@ -1,75 +1,130 @@
-
-        const data = await page.evaluate(() => {
-            const rawData = JSON.parse(document.querySelector('body').innerText);
-            return rawData.items.slice(0, 1).map(item => ({
-                // tutaj jakie dane chcemy aby program zwrocil, bedzie dodatkowy plik ze wszytskim co ma api do zaoferowania
-                title: item.title,
-                total_item_price: item.total_item_price,
-                brand_title: item.brand_title,
-                url: item.url,
-                photo: {
-                    url: item.photo.url
-                }
-            }));
-        });
-
-        if (data.length === 0) {
-            break; // No more items to fetch
+
+async function getCookie(url = vintedUrl){
+    const req = await fetch(url);
+    const cookies = await req.headers.get('set-cookie');
+    const cookie = /_vinted_fr_session=([^;]+)/.exec(cookies)?.[1]
+    return cookie
+}
+
+async function getData(url = dataUrl){
+    const req = await fetch(url, {
+        headers: {
+            cookie: `_vinted_fr_session=${await getCookie()}`
         }
+    })
+    const res = await req.json();
+    return res?.items
+}
+
+async function scrape(){
+    const fetchedOffers = await getData();
+    const cachedOffers = getCachedOffers();
+    const newOffers = getNewOffers(cachedOffers, fetchedOffers);
+    const addedOffers = logOffers(cachedOffers, newOffers);
+    if(addedOffers.length != 0){
+        console.log(`Found ${addedOffers.length}`)
+        console.table(addedOffers.map(x => ({
+            title: x.title,
+            price: x.total_item_price,
+            url: x.url,
+        })))
+    } 
+
+    return addedOffers
+}
 
-        const item = data[0];
 
-        if (!seenItems.has(item.url)) {
-            console.log(item);
-            seenItems.add(item.url);
+function startScraping(){
+    scrape().then(res => {
+        discordSendMsg(res)
+    })
+}
+
+function discordSendMsg(newOffers){
+    if(newOffers.length == 0) return;
+
+    const client = new Discord.Client({
+        intents: ['MessageContent', 'Guilds', 'GuildMessages']
+    });
+
+
+    client.on('ready', () => {
+
+        console.log(`Logged in as ${client.user.tag}`);
+        const channelId = Config.discord.channelID;
+        const channel = client.channels.cache.get(channelId);
+        if (channel) {
+            newOffers.map(offer => channel.send({
+                embeds: [
+                    {
+                        "title": offer.title,
+                        "url": offer.url,
+                        "description": `
+                            cena: **${offer.total_item_price}** ${offer.currency}\n   
+                            marka: ${offer.brand}
+                        `,
+                        "color": 3387490,
+                        "thumbnail": {
+                            "url": offer?.photo?.url
+                        }
+                    }
```

Comment:
to jest json embeds z discorda, musisz w dokumentacji discorda zobaczyc jak sobie to zmienaic, on sluzy do tego jak wyglada wiadomosc na czacie